test/decoder.spec.js

```diff
@@ -1,34 +1,49 @@
+import path from 'path';
+import fs from 'fs';
 import { expect } from 'chai';
-import { decodeInputData } from '../src/decoder';
+import { decodeInputData, decodelogData } from '../src/decoder';
+
 
 describe('decodeInputData', () => {
-    
-    let transactionSample = {};
-    
-    beforeEach(() => {
-        transactionSample = {
-            hash: '0x9fc76417374aa880d4449a1f7f31ec597f00b1f6f3dd2d66f4c9c6c445836d8b',
-            nonce: 2,
-            blockHash: '0xef95f2f1ed3ca60b048b4bf67cde2195961e0bba6f70bcbea9a2c4e133e34b46',
-            blockNumber: 3,
-            transactionIndex: 0,
-            from: '0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b',
-            to: '0x6295ee1b4f6dd65047762f924ecd367c17eabf8f',
-            value: 123,
-            gas: 314159,
-            gasPrice: 123,
-            input: '0x57cb2fc4',
-        };
-    })
+  const esopjson = path.join(__dirname, 'mockedData', 'esopABI.json');
+  const abi = JSON.parse(fs.readFileSync(esopjson, { encoding: 'utf8' }));
+
+  it('should return function name and its params for the transaction', () => {
+    const dataSample1 = '0xcc96b9430000000000000000000000006a58a4e262b6ead9eb40f41313d6d1fbbbb41c160000000000000000000000000000000000000000000000000000000057fabde0000000000000000000000000000000000000000000000000000000005960cc0600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001';
+    const openESOP = { name: 'offerOptionsToEmployee',
+      params:
+     [{ name: 'e',
+       value: '0x6a58a4e262b6ead9eb40f41313d6d1fbbbb41c16',
+       type: 'address' },
+     { name: 'issueDate', value: '1476050400', type: 'uint32' },
+     { name: 'timeToSign', value: '1499515910', type: 'uint32' },
+     { name: 'extraOptions', value: '0', type: 'uint32' },
+     { name: 'poolCleanup', value: true, type: 'bool' }] };
+    const decodedInput = decodeInputData(dataSample1, abi);
+    expect(decodedInput).to.deep.equal(openESOP);
+  });
 
-    it('should return the function name and its params for the transaction', () => {        
-        const decodedInput = decodeInputData(transactionSample);
-        expect(decodedInput).to.contains.keys(['function', 'params']);
-    });
+  it('should throw if input data doesnt decode', () => {
+    const dataSampleErr = 'a58a4e262b6ead9eb4';
+    expect(() => decodeInputData(dataSampleErr, abi)).to.throw('Problem with input data');
+  });
 
-    it('should return the function name and its params for the transaction', () => {    
-        delete transactionSample['input'];
-        const decodedInput = decodeInputData(transactionSample);
-        expect(decodedInput).to.be.equal(null);
-    });
+  it('should return function name and its params for the transaction', () => {
```

Comment:
I think you forget to change this text, because it's the same like the first one.